### PR TITLE
Feature/s132v5

### PR DESCRIPTION
**Pull request by bihanssen**
Added support for SD_API_V5. 
Removed support for SD_API_V3.
Included updated fix for MTU failure bug.
Updated examples for V2/V5 API.

~~Note: connectivity hex files for V5 needs to be added.~~ Hex files have been added.

### CODE REVIEW

File: src/common/ble_gap_impl.cpp [hunk 1]
```diff
@@ -765,7 +773,12 @@ uint32_t sd_ble_gap_sec_params_reply(adapter_t *adapter,
     
     if (p_sec_keyset)
     {
+#if NRF_SD_BLE_API_VERSION < 4
         std::memcpy(&keyset->keyset, p_sec_keyset, sizeof(ble_gap_sec_keyset_t));
+#else
+        //TODO: copy keyset properly
```
Comment (bencefr) [inline src/common/ble_gap_impl.cpp]: Do you intend to complete the keyset handling here?

Comment (bihanssen) [reply src/common/ble_gap_impl.cpp]: Yes, merge gone bad :confounded: Will fix.

Comment (chunfantasy) [reply src/common/ble_gap_impl.cpp]: Fixed and tested :grin:

Comment (bihanssen) [reply src/common/ble_gap_impl.cpp]: :+1:

**Comment by mrodem** [comment]
Seems that building this branch on Linux fails:

```
$ git fetch
$ git status
On branch feature/s132v5
Your branch is up-to-date with 'origin/feature/s132v5'.
$ cmake -G "Unix Makefiles" && make
...
In file included from /home/mrodem/projects/pc-ble-driver/include/common/internal/ble_common.h:43:0,
                 from /home/mrodem/projects/pc-ble-driver/src/common/ble_gatts_impl.cpp:50:
/home/mrodem/projects/pc-ble-driver/src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/app_ble_gap_sec_keys.h:78:43: error: storage size of ‘m_app_keys_table’ isn’t known
 ser_ble_gap_app_keyset_t m_app_keys_table[];
                                           ^
CMakeFiles/pc_ble_driver_obj_sd_api_v5.dir/build.make:542: recipe for target 'CMakeFiles/pc_ble_driver_obj_sd_api_v5.dir/src/common/ble_gatts_impl.cpp.o' failed
make[2]: *** [CMakeFiles/pc_ble_driver_obj_sd_api_v5.dir/src/common/ble_gatts_impl.cpp.o] Error 1
CMakeFiles/Makefile2:252: recipe for target 'CMakeFiles/pc_ble_driver_obj_sd_api_v5.dir/all' failed
make[1]: *** [CMakeFiles/pc_ble_driver_obj_sd_api_v5.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

Building on master with the same command succeeds.

**Comment by chunfantasy** [comment]
@mrodem I rebuild on Windows and it works fine. I build it on Ubuntu and it has the same fails as posted. Now I am looking into it.

**Comment by chunfantasy** [comment]
Yes! Build tests passed! Happy going home!! :stuck_out_tongue_closed_eyes:

File: src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c [hunk 2]
```diff
@@ -18,8 +45,7 @@
 #include "cond_field_serialization.h"
 #include <string.h>
 
-extern ser_ble_gap_app_keyset_t m_app_keys_table[];
-
+//extern ser_ble_gap_app_keyset_t m_app_keys_table[];
```
Comment (mrodem) [inline src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c]: Remove commented out code.

File: src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c [hunk 3]
```diff
@@ -48,7 +74,7 @@ uint32_t ble_gap_evt_auth_key_request_dec(uint8_t const * const p_buf,
 }
 
 
-extern ser_ble_gap_app_keyset_t m_app_keys_table[];
+//extern ser_ble_gap_app_keyset_t m_app_keys_table[];
```
Comment (mrodem) [inline src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c]: Remove commented out code.

File: src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c [hunk 4]
```diff
@@ -171,12 +197,15 @@ uint32_t ble_gap_evt_lesc_dhkey_request_dec(uint8_t const * const p_buf,
 
     uint8_t ser_data;
     SER_PULL_uint16(&p_event->evt.gap_evt.conn_handle);
-    
+
     // keyset is an extension of standard event data - used to synchronize keys at application
-    ser_ble_gap_app_keyset_t *keyset;
-    err_code = app_ble_gap_sec_context_find(p_event->evt.gap_evt.conn_handle, &keyset);
+    //TODO: Add proper impl for keyset
+    //ser_ble_gap_app_keyset_t *keyset;
```
Comment (mrodem) [inline src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c]: Looks like there is some unfinished work here.

File: src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c [hunk 5]
```diff
@@ -171,12 +197,15 @@ uint32_t ble_gap_evt_lesc_dhkey_request_dec(uint8_t const * const p_buf,
...
+    uint32_t keyset_index;
+    err_code = app_ble_gap_sec_context_find(p_event->evt.gap_evt.conn_handle, &keyset_index);
     SER_ASSERT(err_code == NRF_SUCCESS, err_code);
-    p_event->evt.gap_evt.params.lesc_dhkey_request.p_pk_peer = keyset->keyset.keys_peer.p_pk;
+    //p_event->evt.gap_evt.params.lesc_dhkey_request.p_pk_peer = keyset->keyset.keys_peer.p_pk;
```
Comment (mrodem) [inline src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c]: Remove commented out code.

File: .gitignore [hunk 6]
```diff
@@ -3,7 +3,7 @@ _build
 **/CMakeFiles
 **/CMakeCache.txt
 *.pyc
-sdk/
```
Comment (mrodem) [inline .gitignore]: Is this correct? The sdk directory is generated by scripts in hex/sd_api_vX.

Comment (bihanssen) [reply .gitignore]: No, this is not correct, sdk should still be ignored.

File: src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c (hunk 4 above)
Comment (chunfantasy) [reply src/sd_api_v5/sdk/components/serialization/application/codecs/ble/serializers/ble_gap_evt_app.c]: It looks the same as where I changed.
I can change it but I do not have example to test it
which is for the api function ble_gap_evt_lesc_dhkey_request.

**Pull request merged**